adapters/audienceNetwork/facebook.go

```diff
@@ -115,6 +128,50 @@ func (this *FacebookAdapter) makeAuthID(req *openrtb2.BidRequest) string {
 	return hex.EncodeToString(h.Sum(nil))
 }
 
+func dropSingleElement(extension []byte, elementName string, i interface{}) ([]byte, error) {
+	buf := bytes.NewBuffer(extension)
+	dec := json.NewDecoder(buf)
+	var startIndex int64
+	for {
+		token, err := dec.Token()
+		if err == io.EOF {
+			// io.EOF is a successful end
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		if token == elementName {
+			if dec.More() {
+				err := dec.Decode(&i)
+				if err != nil {
+					return nil, err
+				}
+			}
+			endIndex := dec.InputOffset()
+
+			if dec.More() {
+				for {
+					//structure has more elements, need to find index of comma
+					if extension[endIndex] == []byte(",")[0] {
```

Comment:
Consider hardcoding the character code for a comma rather than computing it each loop iteration.

Reply:
Done!